core/client/fs/src/main/java/alluxio/client/file/FileSystemContextReinitializer.java

```diff
@@ -55,9 +57,7 @@
   public FileSystemContextReinitializer(FileSystemContext context) {
     mContext = context;
     mExecutor = new ConfigHashSync(context);
-    mExecutorService = Executors.newSingleThreadExecutor(ThreadFactoryUtils.build(
-        "config-hash-master-heartbeat-%d", true));
-    mExecutorService.submit(
+    REINIT_EXECUTOR.submit(
         new HeartbeatThread(HeartbeatContext.META_MASTER_CONFIG_HASH_SYNC, mContext.getId(),
```

Comment:
I suspect the first submission will hold the thread and rest will never have a chance to run.

Reply:
that makes sense, i'll change it to a scheduled fixed rate executor service and refactor the hearbeat executor